File: test/kine_p560s.py
""" Robotics Toolbox for Python -- Kinematics tests for Puma 560 (std)
"""

import _robot
from robot.testparser import *


tests = '''
echo off
from robot.puma560 import *
echo on

p560

# at nominal pose
arg2array(qn)
t = fkine(p560, qn)
q = ikine560(p560, t)
fkine(p560, q)
ikine(p560, t, q0=[0, 0.7, 3, 0, 0.7, 0], m=[1, 1, 1, 1, 1, 1], verbose=2)
'''

if __name__ == "__main__" :

    testparser(tests)
    

File: Octave/rvctools/robot/tests/kine_p560s.m
% Test inverse kinematics for Puma 560 standard

echo off

mdl_puma560;
p560

echo on

% at nominal pose
qn
t = fkine(p560, qn)
q = ikine6s(p560, t)
fkine(p560, q)
ikine(p560, t, [0, 0.7, 3, 0, 0.7, 0], [1, 1, 1, 1, 1, 1], 'verbose=2')

echo off

